QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,135 @@
+
+### Copyright Statement [3.iv]
+
+The copyright holders each provide a statement of copyright in each source code file in `class_loader`.
+
+There is an automated test which runs a linter (ament_copyright) that ensures each file has at least one copyright statement.
+
+## Testing [4]
+
+### Feature Testing [4.i]
+
+Each feature in `class_loader` has corresponding tests which simulate typical usage, and they are located in the `test` directory.
+New features are required to have tests before being added.
+
+### Public API Testing [4.ii]
+
+Each part of the public API have tests, and new additions or changes to the public API require tests before being added.
+The tests aim to cover both typical usage and corner cases, but are quantified by contributing to code coverage.
+
+### Coverage [4.iii]
+
+`class_loader` follows the recommendations for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#coverage), and opts to use branch coverage instead of line coverage.
+
+This includes:
+
+- tracking and reporting branch coverage statistics
+- achieving and maintaining branch coverage at or above 95%
+- no lines are manually skipped in coverage calculations
+
+Changes are required to make a best effort to keep or increase coverage before being accepted, but decreases are allowed if properly justified and accepted by reviewers.
+
+Current coverage statistics can be viewed here:
+
+![](https://codecov.io/gh/ahcorde/class_loader/branch/ahcorde_code_coverage/graphs/sunburst.svg)
+
+### Performance [4.iv]
+
+`class_loader` follows the recommendations for performance testing of C++ code in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#performance-testing), and opts to do performance analysis on each release rather than each change.
+
+### Linters and Static Analysis [4.v]
+
+`class_loader` uses and passes all the standard linters and static analysis tools for a C++ package as described in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#linters-and-static-analysis).
+
+## Dependencies [5]
+
+### Direct Runtime ROS Dependencies [5.i]
+
+### Optional Direct Runtime ROS Dependencies [5.ii]
+
+### Direct Runtime non-ROS Dependency [5.iii]
+
+`class_loader` has run-time and build-time dependencies that need to be considered for this declaration.
+
+ - console_bridge_vendor
+ - libconsole-bridge-dev
+
+It has one "buildtool" dependency, which do not affect the resulting quality of the package, because they do not contribute to the public library API.
+It also has several test dependencies, which do not affect the resulting quality of the package, because they are only used to build and run the test code.
+
+## Platform Support [6]
+
+`class_loader` supports all of the tier 1 platforms as described in [REP-2000](https://www.ros.org/reps/rep-2000.html#support-tiers), and tests each change against all of them.
```

Comment:
Add links to latest CI results. and remove TODO. I don't think we need to say anything about non-tier 1 unless we have ci tests for them.